src/main/java/io/cdap/plugin/marketo/source/batch/MarketoReportingSourceConfig.java

```diff
@@ -0,0 +1,187 @@
+    if (schema == null) {
+      schema = MarketoReportingSchemaHelper.getSchema(this);
+    }
+    return schema;
+  }
+
+  public Marketo getMarketo() {
+    if (marketo == null) {
+      marketo = new Marketo(getRestApiEndpoint(), getClientId(), getClientSecret());
+    }
+    return marketo;
+  }
+
+
+  public String getClientId() {
+    return clientId;
+  }
+
+  public String getClientSecret() {
+    return clientSecret;
+  }
+
+  public String getRestApiEndpoint() {
+    return restApiEndpoint;
+  }
+
+  public String getStartDate() {
+    return startDate;
+  }
+
+  public String getEndDate() {
+    return endDate;
+  }
+
+  public ReportType getReportType() {
+    return ReportType.fromString(reportType);
+  }
+
+  void validate(FailureCollector failureCollector) {
+    validateDate(failureCollector);
+    validateReportType(failureCollector);
+    validateMarketoEndpoint(failureCollector);
+    validateSecrets(failureCollector);
+  }
+
+  void validateDate(FailureCollector failureCollector) {
+    if (!(containsMacro(PROPERTY_START_DATE) && containsMacro(PROPERTY_END_DATE))) {
+      try {
+        Helpers.getDateRanges(getStartDate(), getEndDate());
+      } catch (IllegalArgumentException ex) {
+        String message = String.format("Failed to validate dates: %s.", ex.getMessage());
+        String correctiveAction = null;
+        if (ex.getMessage().contains("start date more than end date")) {
+          message = "Start date more than end date.";
+          correctiveAction = "Swap dates.";
+        }
+        failureCollector.addFailure(message, correctiveAction)
+          .withConfigProperty(PROPERTY_START_DATE).withConfigProperty(PROPERTY_END_DATE);
+      } catch (DateTimeParseException ex) {
+        failureCollector.addFailure("Failed to parse one of dates.",
+                                    "Correct dates to ISO 8601 format.")
+          .withConfigProperty(PROPERTY_START_DATE).withConfigProperty(PROPERTY_END_DATE);
+      }
+    }
+  }
+
+  void validateReportType(FailureCollector failureCollector) {
+    if (!containsMacro(PROPERTY_REPORT_TYPE)) {
+      try {
+        getReportType();
+      } catch (IllegalArgumentException ex) {
+        failureCollector.addFailure(String.format("Incorrect reporting type '%s'.", reportType),
+                                    "Set reporting type to 'activities' or 'leads'.")
+          .withConfigProperty(PROPERTY_REPORT_TYPE);
+      }
+    }
+  }
+
+  void validateSecrets(FailureCollector failureCollector) {
+    if (!containsMacro(PROPERTY_CLIENT_ID) && Strings.isNullOrEmpty(getClientId())) {
+      failureCollector.addFailure("Client ID is null or empty.",
+                                  "Set Client ID to non empty string.")
+        .withConfigProperty(PROPERTY_CLIENT_ID);
+    }
+
+    if (!containsMacro(PROPERTY_CLIENT_SECRET) && Strings.isNullOrEmpty(getClientSecret())) {
+      failureCollector.addFailure("Client Secret is null or empty.",
```

Comment:
Let's not use words like null or something. This are messages shown to client.
"Client Secret should not be empty"